internal/core/lib.rs

```diff
@@ -18,7 +18,7 @@ pub mod unsafe_single_core {
     macro_rules! thread_local_ {
         ($(#[$($meta:tt)*])* $vis:vis static $ident:ident : $ty:ty = $expr:expr) => {
             $(#[$($meta)*])*
-            pub(crate) static $ident: crate::unsafe_single_core::FakeThreadStorage<$ty> = {
+            $vis static $ident: crate::unsafe_single_core::FakeThreadStorage<$ty> = {
```

Comment:
I think the reason why i didn't want to allow public stuff there is because this is not sound, and i wanted to keep it local to the crate...
But that said, i guess we don't have much choice if we need to use it in the backend.